### PR TITLE
CI housekeeping

### PR DESCRIPTION
**Pull request by kianmeng**
List of changes:
- add Elixir 1.17.x / Erlang OTP 27.x to CI
- bump github/actions
- bump min support versions to Elixir 1.6.x / Erlang OTP 21
- use ubuntu-20.04 as OS image due to depreciation of 18.04

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,29 +8,29 @@ on:
 
 jobs:
   test:
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-20.04
```
Comment (kianmeng) [inline .github/workflows/ci.yml:11]: > The ubuntu-18.04 environment is deprecated, consider switching to ubuntu-20.04(ubuntu-latest), or ubuntu-22.04 instead. For more details see https://github.com/actions/virtual-environments/issues/6002

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -8,29 +8,29 @@ on:
...
       MIX_ENV: test
     strategy:
       fail-fast: false
       matrix:
         include:
           - pair:
-              elixir: 1.5.3
-              otp: 19.3.6.13
+              elixir: 1.6.x
+              otp: 21.x
```
Comment (kianmeng) [inline .github/workflows/ci.yml:20]: I bumped the min support Elixir version to 1.6, as OTP 21 is the minimum supported version in Ubuntu 20.04 OS image, see https://github.com/erlef/setup-beam?tab=readme-ov-file#compatibility-between-operating-system-and-erlangotp

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Comment by kianmeng** [comment]
:partying_face: :partying_face: :partying_face: :partying_face: :partying_face:

**Pull request merged**